runtime/src/main/java/org/corfudb/runtime/collections/TxBuilder.java

```diff
@@ -0,0 +1,270 @@
+    }
+
+    private <M extends Message> void validateVersion(@Nullable M previousMetadata,
+                                                     @Nullable M userMetadata) {
+        if (userMetadata == null) {
+            return;
+        }
+        for (Descriptors.FieldDescriptor fieldDescriptor : userMetadata.getDescriptorForType().getFields()) {
+            if (fieldDescriptor.getOptions().getExtension(CorfuOptions.schema).getVersion()) {
+                if (!versionTypes.contains(fieldDescriptor.getType())) {
+                    throw new IllegalArgumentException("Version field needs to be an Integer or Long type.");
+                }
+                long validatingVersion = (long) userMetadata.getField(fieldDescriptor);
+                long previousVersion = Optional.ofNullable(previousMetadata)
+                        .map(m -> (Long) m.getField(fieldDescriptor))
+                        .orElse(-1L);
+                if (validatingVersion != previousVersion) {
+                    throw new RuntimeException("Stale object Exception");
+                }
+            }
+        }
+    }
+
+    private <M extends Message> M getNewMetadata(@Nonnull M previousMetadata,
+                                                 @Nullable M userMetadata) {
+        M.Builder builder = previousMetadata.toBuilder();
+        for (Descriptors.FieldDescriptor fieldDescriptor : previousMetadata.getDescriptorForType().getFields()) {
+            if (fieldDescriptor.getOptions().getExtension(CorfuOptions.schema).getVersion()) {
+                builder.setField(
+                        fieldDescriptor,
+                        Optional.ofNullable(previousMetadata.getField(fieldDescriptor))
+                                .map(previousVersion -> ((Long) previousVersion) + 1)
+                                .orElse(0L));
+            } else if (userMetadata != null){
+                builder.setField(fieldDescriptor, userMetadata.getField(fieldDescriptor));
+            }
+        }
+        return (M) builder.build();
+    }
+
+    private <K extends Message, V extends Message, M extends Message>
+    Table<K, V, M> getTable(@Nonnull final String tableName) {
+        return this.tableRegistry.getTable(this.namespace, tableName);
+    }
+
+    /**
+     * Creates a record for the specified key.
+     *
+     * @param tableName Table name to perform the create on.
+     * @param key       Key.
+     * @param value     Value to create.
+     * @param <K>       Type of Key.
+     * @param <V>       Type of Value.
+     * @return TxBuilder instance.
+     */
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    TxBuilder create(@Nonnull final String tableName,
+                     @Nonnull final K key,
+                     @Nonnull final V value,
+                     @Nullable final M metadata) {
+        Table<K, V, M> table = getTable(tableName);
+        validateNamespace(table);
```

Comment:
Why `validateNamespace` is  needed here? When calling getTable, we use the `namespace` in this class, and find the table registered by this namespace, unless there is a bug in tableRegistry, right?

Reply:
validateNamespace is used in case someone tries to start a transaction on one namespace but adds in a table which belongs to a different namespace.
It's to catch a user error.